includes/class-wc-ajax.php

```diff
@@ -1910,6 +1910,138 @@ public static function save_variations() {
 		wp_die();
 	}
 
+	/**
+	 * Reduce order item stock.
+	 *
+	 * @since 3.5.3
+	 */
+	public static function reduce_order_item_stock() {
+		check_ajax_referer( 'order-item', 'security' );
+		if ( ! current_user_can( 'edit_shop_orders' ) ) {
+			wp_die( -1 );
+		}
+
+		$order_id       = absint( $_POST['order_id'] );
+		$order_item_ids = isset( $_POST['order_item_ids'] ) ? $_POST['order_item_ids'] : array();
+		$order          = wc_get_order( $order_id );
+		$order_items    = $order->get_items();
+		$return         = array();
+
+		if ( $order && $order->allow_manual_stock_adjustment() && ! empty( $order_items ) && sizeof( $order_item_ids ) > 0 ) {
+			foreach ( $order_items as $item_id => $order_item ) {
+				// Only reduce checked items
+				if ( ! in_array( $item_id, $order_item_ids ) ) {
+					continue;
+				}
+
+				$_product = $order_item->get_product();
+				if ( $_product && $_product->exists() && $_product->managing_stock() ) {
+					$net_stock_change = $order_item->get_quantity();
+					$reduced = $order_item->get_meta( '_reduced_stock', true );
+					if ( ! $reduced ) {
+						$reduced = $net_stock_change;
+					} elseif ( $net_stock_change <= 								$reduced ) {
```

Comment:
Why so much whitespace?

Reply:
Thanks, I'm not sure why phpcs didn't catch that one.